### PR TITLE
Use flask blueprints to cleanup routing code

### PR DESCRIPTION
**Pull request by lukeyeager**
> Blueprints can greatly simplify how large applications work and provide a central means for Flask extensions to register operations on applications.
http://flask.pocoo.org/docs/0.10/blueprints/

This just helps with code quality and readability. This doesn't make any changes that are externally visible in the application.
* Add routes deliberately in `webapp.py` rather than implicitly through each of the `views.py` files
* Makes the `url_for()` statements match the actual code layout more closely

### CODE REVIEW

File: digits/dataset/images/classification/test_views.py [hunk 1]
```diff
@@ -129,9 +129,10 @@ def create_dataset(cls, **kwargs):
             s = BeautifulSoup(rv.data, 'html.parser')
             div = s.select('div.alert-danger')
             if div:
-                raise RuntimeError(div[0])
+                print div[0]
```
Comment (gheinrich) [inline digits/dataset/images/classification/test_views.py]: is this related to this change?

Comment (lukeyeager) [inline digits/dataset/images/classification/test_views.py]: You're right, it wasn't. New PR at #509.

**Comment by gheinrich** [comment]
All the routes I tried worked for me, including a number of JSON routes.
